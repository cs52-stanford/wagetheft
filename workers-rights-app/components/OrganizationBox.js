import React, { Component } from "react";
import {
    View,
    Text,
    StyleSheet,
    Image,
    TouchableOpacity,
    Dimensions
} from "react-native";
import Colors from '../constants/Colors';

const OrganizationBox = props => {
    return (
        <TouchableOpacity onPress={props.onSelect} style={styles.box}>
                <View style={{ flex: 4 }}>
                    <Image source={{uri: props.image}} style={styles.imgStyle}/>
                </View>
                <View style={{ flex: 1, justifyContent: 'center', marginLeft: 25}}>
                    <Text numberOfLines={2} style={styles.orgTitle}>{props.title}</Text>
                </View>
        </TouchableOpacity>
    )
};

const styles = StyleSheet.create({
    box: {
        height: 229,
        width: 284,
        marginRight: 15,
        borderRadius: 6,
        shadowOpacity: 0.25,
        shadowColor: "black",
        shadowOffset: { width: 0, height: 2 },
        shadowRadius: 4,
        elevation: 5,
    },
    orgTitle: {
        fontSize: 16,
        fontFamily: 'nunito-semibold'
    },
    imgStyle: {
        flex: 1, 
        width: null,
        height: null, 
        resizeMode: 'cover',
        borderRadius: 6
    }
});

export default OrganizationBox;