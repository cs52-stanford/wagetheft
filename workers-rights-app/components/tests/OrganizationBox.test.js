import React from "react";
import OrganizationBox from "../OrganizationBox";
import { render, fireEvent } from "@testing-library/react-native";

describe("<OrganizationBox /> Tests", () => {
  it("should match snapshot", () => {
    const mockPressFunction = jest.fn();
    const rendered = render(
      <OrganizationBox
        title="Test Title"
        onSelect={mockPressFunction}
        image="1"
      />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it("should display text properly", () => {
    const mockPressFunction = jest.fn();
    const rendered = render(
      <OrganizationBox
        title="Test Title"
        onSelect={mockPressFunction}
        image="1"
      />
    );
    const textComponent = rendered.getByTestId("text");
    expect(textComponent.props.children).toEqual("Test Title");
  });

  it("should fire onPress events", () => {
    const onSelect = jest.fn();
    const rendered = render(
      <OrganizationBox title="Test Title" onSelect={onSelect} image="1" />
    );
    const buttonComponent = rendered.getByTestId("orgBox");
    fireEvent(buttonComponent, "press");
    expect(onSelect).toHaveBeenCalled();
  });
});
