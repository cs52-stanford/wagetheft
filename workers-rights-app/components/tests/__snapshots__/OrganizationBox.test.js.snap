// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrganizationBox /> Tests should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 229,
      "marginRight": 15,
      "opacity": 1,
      "width": 284,
    }
  }
  testID="orgBox"
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "elevation": 5,
        "flex": 4,
        "margin": "2%",
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "1",
        }
      }
      style={
        Object {
          "borderRadius": 6,
          "flex": 1,
          "height": null,
          "resizeMode": "cover",
          "width": null,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 25,
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        Object {
          "fontFamily": "nunito-semibold",
          "fontSize": 16,
        }
      }
      testID="text"
    >
      Test Title
    </Text>
  </View>
</View>
`;
